### ISSUE TITLE
Add onTapDown to DataRow2

### ISSUE THREAD

**Issue by JerwuQu** [maxim-saplin/data_table_2#97]
`onTapDown` on `DataRow2` would be useful to get touch position for creating menus on `onTap` or `onLongPress`.

`DataRow2` also already has `onSecondaryTapDown`, so I feel it would make sense to also have `onTapDown`.

I tried to do a workaround by wrapping the whole `DataTable2` within a `GestureDetector` with an `onTapDown` for now, but `DataTable2` would end up getting priority so it didn't work.

**Comment by maxim-saplin** [comment]
Hi @JerwuQu . I've added row level Down, Up and Cancel events, thought they seem inconsistent (e.g. tap down happening only after a sligh delay, quick tap right away fires row tap ignoring tap down). You can try `more-row-tap-events` branch

**Comment by JerwuQu** [comment]
Hello! I see, that's strange... 

I took a look at the branch but it looks to be identical to `main` right now. Commits not pushed yet perhaps?
https://github.com/maxim-saplin/data_table_2/commits/main
https://github.com/maxim-saplin/data_table_2/commits/more-row-tap-events

**Comment by maxim-saplin** [comment]
Sorry, forgot to push the commit - please check now

**Comment by JerwuQu** [comment]
Good morning 🌞 

As you said, the `onTapDown` doesn't run if you tap too quickly, which is the same behavior I got when I wrapped the whole `DataTable2` in a `GestureDetector`. I am pretty sure I got it working with the flutter standard `DataTable`, but now I'm not so sure anymore. I can test it more after work.

Regardless, it works in conjunction with `onLongPress` which is still useful, and is enough for my project right now.

About `...TapUp` and `...TapCacel`, I'm not sure what they would be used for, so I can't test in a realistic context.

**Comment by maxim-saplin** [comment]
I didn't give the event system in Flutter a thorough look, though it seems that there're gestures and pointer events - the first is a more abstract concept and handles by geasture detectors, while the second one are about raw pointer events and are handleled by listeners. Flutter docs don't give much of explanation (https://docs.flutter.dev/development/ui/advanced/gestures), 20 minute research didn't prove answers to tap, tapDown, tapUp gesture odd behaviour.

Keeping the issue open and I won't push the change to main for the time being

**Comment by JerwuQu** [comment]
I did some more experimenting myself and I don't know why it happens either, but it turns out `onPointerDown` from `Listener` works for my case, so you could either:
- Close this issue (since it works on my end now)
- Implement `onTapDown` yourself (using a `Listener` and `onPointerDown`) for user convenience
- Leave the issue up as wontfix and hope something changes in the future so the branch can be merged as-is

**Comment by maxim-saplin** [comment]
Thanks for the feedback, let's keep the issue open

**Comment by martinogarrizzo5** [comment]
Hi there! Are there any updates about this feature? I would be interested in using it as I would like to show a menu on the row position after a long press
